### PR TITLE
added support to emulate device viewport

### PR DESCRIPTION
**Pull request by saikrishna321**
This PR will help users to emulate device viewport like iPhone 6, iPad, Nexus and more.

Code snippet 

```
(async () => {
    try {
        await openBrowser();
        await goto('https://taiko.gauge.org/');
        await emulateDevice('iPhone 6');
        await evaluate(() => window.innerWidth);
        await evaluate(() => window.innerHeight);
    } catch (e) {
        console.error(e);
    } finally {
        await closeBrowser();
    }
})();
```
List of devices: https://github.com/getgauge/taiko/blob/5d7524b90b48b12ce3d58dc52f54e36ca9ab97ca/lib/device.js

screen shot 2019-01-24 at 9 26 42 pm

### CODE REVIEW

**Comment by zabil** [comment]
Amazing! I know a lot of people waiting for this feature.

Can we also make this similar to `emulateNetwork` ?
https://github.com/getgauge/taiko/pull/340#issuecomment-453901645

And pass it via the command line in addition to the API?

**Comment by saikrishna321** [comment]
> pass it via the command line in addition to the API?

@zabil 

Yes, i can make the changes `taiko test.js emulateDevice='iPhone 6'`

**Comment by saikrishna321** [comment]
@zabil I have added support from commandline

`
 taiko test.js --emulateDevice='iPhone 6'
`

File: lib/taiko.js [hunk 1]
```diff
@@ -112,7 +113,9 @@ module.exports.openBrowser = async (options = { headless: true }) => {
     currentHost = endpoint.host;
     currentPort = endpoint.port;
     await connect_to_cri();
-    return { description: 'Browser opened' };
+    const device = commandlineArgs().emulateDevice;
```
Comment (NivedhaSenthil) [inline lib/taiko.js]: The command line logic can go to the runner, and can be accessed as Env variable ( TAIKO_EMULATE_DEVICE )from API.

Comment (saikrishna321) [reply lib/taiko.js]: I have address as per @zabil comment https://github.com/getgauge/taiko/pull/340#issuecomment-454652434, so should i remove from commandLineArg and just add ENV access ?

Comment (NivedhaSenthil) [reply lib/taiko.js]: we will need both `taiko script.js --emulateDevice=iPhone` and `TAIKO_EMULATE_DEVICE=iphone gauge run specs` when using other runners

Comment (saikrishna321) [reply lib/taiko.js]: Done

**Comment by saikrishna321** [comment]
@NivedhaSenthil fixed comments. Pls review

File: lib/taiko.js [hunk 2]
```diff
@@ -112,7 +113,9 @@ module.exports.openBrowser = async (options = { headless: true }) => {
     currentHost = endpoint.host;
     currentPort = endpoint.port;
     await connect_to_cri();
-    return { description: 'Browser opened' };
+    const device = process.env.TAIKO_EMULATE_DEVICE || commandlineArgs().emulateDevice;
```
[the remark this answers is not in the document]
Comment (saikrishna321) [reply lib/taiko.js]: Yes, that's exactly what I have addressed `TAIKO_EMULATE_DEVICE=iphone gauge run specs
` with this command the view port will be emulated with respective value. So when user wants emulate the device port on REPL they can send taiko location.js --emulateDevice='iPhone 6'

**Comment by saikrishna321** [comment]
@NivedhaSenthil i have made --emulateDevice as part of taiko runner  

` taiko script.js --emulateDevice='iPhone 6' `

Other runners(jest/gauge/mocha) can access the API directly for ex: 

`
TAIKO_EMULATE_DEVICE='iPhone 6' gauge run specs
emulateDevice(process.env.TAIKO_EMULATE_DEVICE)
`

**Pull request merged**